tools/common-code/cmonitor_watcher.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/python3
+
+#
+# cmonitor_watcher.py
+#
+# Author: Satyabrata Bharati
+# Created: April 2022
+#
+import inotify.adapters
+import queue
+import os
+import time
+import logging
+from datetime import datetime
+
+exit_flag = False
```

Comment:
this flag should not be here... this class should not care about the whole app being requested to exit: it must just process the inotify events and fill a queue as fast as possible...that's it